lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs

```diff
@@ -157,22 +157,58 @@ public void DeleteDirectory(Uri directoryUri)
         }
 
         /// <summary>
-        /// Gets the children of the blob "directory."
+        /// Gets the children of the container (if the uri has segments)
+        /// or containers in the storage account (if the uri has no segments)
         /// </summary>
         public IEnumerable<Uri> GetChildren(Uri directoryUri)
         {
+            string[] segments = directoryUri.Segments;
             BlobContinuationToken blobContinuationToken = null;
-            var path = directoryUri.AbsolutePath.Trim('/');
+
+            // If at the root, return all containers
+            if (segments.Count() <= 1)
```

Comment:
better use `segments.Length` instead of Linq

Reply:
Fixed.

---
In reply to: [196652155](https://github.com/apache/reef/pull/1464#discussion_r196652155) [](ancestors = 196652155)